### PR TITLE
Add support for reading and writing `CF_BITMAP` images

### PR DESCRIPTION
**Pull request by matheuslessarodrigues**
I tried to change as little as possible. However, with the code I had, I could not keep the same interface as the `Bitmap` struct in `image.rs`. So I just commented out the previous code and I tried to keep things simple by only exposing a `Vec<u8>`, which is the raw bytes from a `.bmp` image file.

From the test I added, it seems to work fine. Fell free to use the code in this PR as you want.

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -1,24 +1,254 @@
...
+
+        if unsafe {
+            GetObjectW(
+                handle.as_ptr(),
+                mem::size_of::<BITMAP>() as _,
+                &mut bitmap as *mut BITMAP as _,
+            )
+        } == 0
+        {
+            return Err(io_error("!GetObjectW"));
```
Comment (DoumanAsh) [inline src/image.rs:95]: Use [last_os_error](https://doc.rust-lang.org/std/io/struct.Error.html#method.last_os_error) any time winapi call fails

File: src/image.rs [hunk 2]
```diff
@@ -1,24 +1,254 @@
...
+    pub bytes: Vec<u8>,
+}
+
+fn io_error(message: &'static str) -> io::Error {
+    io::Error::new(io::ErrorKind::Other, message)
+}
+
+impl Image {
+    pub(crate) fn from_handle(handle: ptr::NonNull<c_void>) -> io::Result<Self> {
+        let mut bitmap = BITMAP {
```
Comment (DoumanAsh) [inline src/image.rs:77]: `let mut bitmap: BITMAP = unsafe { mem::MaybeUninit::zeroed().assume_init() } `

File: src/image.rs [hunk 3]
```diff
@@ -1,24 +1,254 @@
 //! Image module
 
-use winapi::um::wingdi::{GetDIBits, GetObjectW};
+use std::{
```
Comment (DoumanAsh) [inline src/image.rs:3]:
```
use std::io::{self, Cursor, Write};
use core::{mem, ptr, slice};
use core::ops::{Deref, DerefMut};
use core::ffi::c_void
```

File: tests/clipboard.rs [hunk 4]
```diff
@@ -105,7 +112,7 @@ fn set_data() {
 
     let expected_text = "For my waifu!";
     let text = "For my waifu!\0gg"; //Pass some garbage after \0
-    //Check set of wide utf-8 bytes
+                                    //Check set of wide utf-8 bytes
```
Comment (DoumanAsh) [inline tests/clipboard.rs:115]: Please remove extra spaces

File: src/image.rs [hunk 5]
```diff
@@ -1,24 +1,254 @@
...
 }
+impl<T> DerefMut for LocalMemory<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { self.0.as_mut() }
+    }
+}
+
+///Bitmap image from clipboard
+pub struct Image {
+    ///Raw image data
```
Comment (DoumanAsh) [inline src/image.rs:67]: Please elaborate on image format (i.e. as example that it is `BMP` image)

File: src/image.rs (hunk 2 above)
Comment (matheuslessarodrigues) [reply src/image.rs:77]: but why would we add unecessary unsafe there? why is this preferred instead of simply creating the struct?

Comment (DoumanAsh) [reply src/image.rs:77]: well it shortens code, but if you don't want to then it is fine :)

**Comment by matheuslessarodrigues** [comment]
I'm doing these changes right now. Should I delete the old code in the coments then?

**Comment by DoumanAsh** [comment]
@matheuslessarodrigues yes sure, now that we have your work there is no need to keep old code

**Comment by matheuslessarodrigues** [comment]
just updated the pr with the changes we discussed!

**Comment by matheuslessarodrigues** [comment]
yeah! I tested on paint, firefox (pasting on twitter, slack, etc) and paint.net

**Comment by DoumanAsh** [comment]
Great, thank I'll release these changes soon

**Pull request merged**